src/services/reports.js

```diff
@@ -24,6 +26,15 @@ const createHandler = async (req, res) => {
     });
 
     await report.save();
+    
+    if(report.type==="no-settlement"){
+      const room = await roomModel.findById(roomId);
+      const emailRoomName = room ? room.name : "";
+      const emailRoomId = room ? room._id : "";
+      const emailHtml = emailPage(reported.name, reported.nickname, emailRoomName, user.nickname, emailRoomId)
+      sendReportEmail(user.nickname, reported.email, report, emailHtml);
+    }
+
```

Comment:
`room`을 핸들러 상단에서 먼저 찾고 가는 로직도 괜찮을 것 같아요 (??)

```suggestion
    const room = await roomModel.findById(roomId);
    if (!room) return res.status(400).json({ error: "User/report: no corresponding room" });
    
    if (report.type === "no-settlement" && room) {
      const emailHtml = emailPage(reported.name, reported.nickname, room.name, user.nickname, roomId)
      sendReportEmail(user.nickname, reported.email, report, emailHtml);
    }

```

Reply:
이 부분도 고려했는데 위에서 말한 것처럼 roomId가 option으로 정의돼서 위에 로직을 사용했습니다. 어차피 front 코드 변경되면 roomId가 사실상 required가 될 것 같은데 아래 로직으로 바꿀까요??

Reply:
`required` 로 바꾸로 front와 함께 같이 업데이트 버전 배포하면 좋을 것 같습니다 
(front 코드도 바뀌어야 하기에 숙지하고 있겠습니다)

Reply:
앗 알겠습니다!! roomId도 required로 수정할게요~!
그렇다면 기존 db migration도 해야 되는 거지요?

Reply:
@chlehdwon 넵넵 마이그레이션 하여도 API 상에서는 `required` 하게 받아야 하니깐 마이그레이션 여부 상관 없이 일단 API에서는 `required`로 받는 것이 좋을 것 같아요

Reply:
@14KGun 일단 validation 스키마에서 required로 변경 해놓았습니다~! front에서 간단하게 작업하고 dev에서 test해보면 좋을 것 같아요